pkg/logs/config/endpoints.go

```diff
@@ -55,6 +56,43 @@ type Endpoint struct {
 	Origin    IntakeOrigin
 }
 
+// GetStatus returns the endpoint status
+func (e *Endpoint) GetStatus(prefix string, useHTTP bool) string {
+	compression := "uncompressed"
+	if e.UseCompression {
+		compression = "compressed"
+	}
+
+	host := e.Host
+	port := e.Port
+
+	var protocol string
+	if useHTTP {
+		if e.UseSSL {
+			protocol = "HTTPS"
+			if port == 0 {
+				port = 443 // use default port
+			}
+		} else {
+			protocol = "HTTP"
+			// this case technically can't happens. In order to
+			// disable SSL, user have to use a custom URL and
+			// specify the port manually.
+			if port == 0 {
+				port = 80 // use default port
+			}
+		}
+	} else {
+		if e.UseSSL {
+			protocol = "SSL encrypted TCP"
+		} else {
+			protocol = "TCP"
+		}
+	}
```

Comment:
It looks like this was originally copy/pasted from `pkg/logs/status/builder.go`.  Perhaps this should be a utility function?  I see that in this PR it's just code moving around, so I suppose it's OK to leave it as-is.

Reply:
What do you mean by `utility function` ?